back-end/yelpCrawler.py

```diff
@@ -0,0 +1,197 @@
+# -*- coding: utf-8 -*-
+from __future__ import print_function
+from os.path import join, dirname
+from dotenv import load_dotenv, find_dotenv
+from bs4 import BeautifulSoup
+import argparse
+import json
+import pprint
+import requests
+import sys
+import urllib
+import os
+
+try:
+    # For Python 3.0 and later
+    from urllib.error import HTTPError
+    from urllib.parse import quote
+    from urllib.parse import urlencode
+except ImportError:
+    # Fall back to Python 2's urllib2 and urllib
+    from urllib2 import HTTPError
+    from urllib import quote
+    from urllib import urlencode
+
+dotenv_path = join(dirname(__file__), '.env')
+load_dotenv(dotenv_path, override=True)
+
+API_KEY = os.environ.get("API_KEY")
+
+# API constants, you shouldn't have to change these.
+API_HOST = 'https://api.yelp.com'
+SEARCH_PATH = '/v3/businesses/search'
+BUSINESS_PATH = '/v3/businesses/'  # Business ID will come after slash.
+
+# Defaults for our simple example.
+DEFAULT_TERM = 'cafes'
+LATITUDE = 37.778281
+LONGITUDE = -122.411865
+RADIUS = 1000
+OPEN_NOW = 'false'
+SEARCH_LIMIT = 10
+
+def request(host, path, api_key, url_params=None):
+    """Given your API_KEY, send a GET request to the API.
+    Args:
+        host (str): The domain host of the API.
+        path (str): The path of the API after the domain.
+        API_KEY (str): Your API Key.
+        url_params (dict): An optional set of query parameters in the request.
+    Returns:
+        dict: The JSON response from the request.
+    Raises:
```

Comment:
Not familiar with python world words, but is it normal to use 'Raises'? I'm more familiar with 'Throws' but up to you :)